drivers/overlay/joinleave.go

```diff
@@ -24,10 +24,25 @@ func (d *driver) Join(nid, eid string, sboxKey string, jinfo driverapi.JoinInfo,
 		return fmt.Errorf("could not find endpoint with id %s", eid)
 	}
 
+	s := n.getSubnetforIP(ep.addr)
+	if s == nil {
+		return fmt.Errorf("could not find subnet for endpoint %s", eid)
+	}
+	// If the subnet doesn't have a vni assigned yet query the datastore first
+	if s.vni == 0 {
```

Comment:
Can we just do this check inside `obtainVxlanID` and return success if vni is non-zero rather than checking this in all places that we call `obtainVxlanID`?